util/db/dbutil.go

```diff
@@ -325,13 +339,25 @@ func (db *Accessor) atomic(fn idemFn, commitLocker sync.Locker, extras ...interf
 		commitLocker.Unlock()
 	}
 
-	end := time.Now()
-	delta := end.Sub(start)
-	if delta > time.Second {
-		db.getDecoratedLogger(fn, extras).Warnf("dbatomic: tx took %v", delta)
-	} else if delta > time.Millisecond {
-		db.getDecoratedLogger(fn, extras).Debugf("dbatomic: tx took %v", delta)
+	if time.Now().After(atomicDeadline) {
+		db.getDecoratedLogger(fn, extras).Warnf("dbatomic: tx surpassed expected deadline by %v", atomicDeadline.Sub(time.Now()))
+	}
+	return
+}
+
+// ResetTransactionWarnDeadline allow the atomic function to extend it's warn deadline by setting a new deadline.
+// The Accessor can be copied and therefore isn't suitable for multi-threading directly,
+// however, the transaction context and transaction object can be used to uniquely associate the request
+// with a particular deadline.
+// the function fails if the given transaction is not on the stack of the provided context.
+func ResetTransactionWarnDeadline(ctx context.Context, tx *sql.Tx, deadline time.Time) (prevDeadline time.Time, err error) {
```

Comment:
is it supposed to be used in subsequent PR?

Reply:
Yes. it will be in the subsequent draft pr..